### DISCUSSION TITLE
How To: Force update database with PhotoPrism\storage\sidecar\...\yml files ?

### DISCUSSION THREAD

**Discussion by pbenaim** [photoprism/photoprism#3883] [Q&A] [answered]
Context:
- I use the Build 231021-9abea5b55 version with Docker Desktop under windows on one "little" server ( Beelink Mini S12 Pro )

- I'm be able to find many descriptions with AI ( LLava [https://github.com/haotian-liu/LLaVA](url) ) this is amazing but we need GPU with minimum 12 Go of memory, so I use it on my personnal computer (  i7 i7-12700KF and NVIDIA 3080 12Go ), so can't work on my PhotoPrism server... 👍 

- After generate Descriptions and Keyswords it's easy to modify all files in PhotoPrism\storage\sidecar\...\*.yml files ( with homemade software 👍 ) on yml Details:

- It's complicated ( for me ) with Docker Desktop to access directely to the PhotoPrism database from another computer ( on same LAN of course). ( execpt ifyou have an easy solution )

My Question:

- How To: Force update database with PhotoPrism\storage\sidecar\...\yml files ? (YAML)

- Of course I would like the faster way...

Perhaps to motivate your answers, I could distribute my in-house software, if you have a graphics card that can run LLaVa...

Thank you again for your help.

**Comment by lastzero** [comment]
YAML files are always just used as export / backup as this would otherwise cause a loop:

- https://docs.photoprism.app/user-guide/backups/export/

JSON files as created by Exiftool (or Google Photos) can be used to read / inject metadata. However, those won't be written and can thus not be used as backups.

**Reply by pbenaim** [reply]
So if I put one json (like https://github.com/photoprism/photoprism/blob/develop/internal/meta/testdata/altitude.json) with same name and in same directory of my picture Photoprism will update datas  ?

**Reply by pbenaim** [reply, marked as answer]
Ok, I done with XMP 👍

**Reply by tilusnet** [reply]
> YAML files are always just used as export / backup as this would otherwise cause a loop:
> 
> * https://docs.photoprism.app/user-guide/backups/export/
> 
> JSON files as created by Exiftool (or Google Photos) can be used to read / inject metadata. However, those won't be written and can thus not be used as backups.

That's interesting to know.

Prove me wrong please, but I sense some contradiction with the [Restoring Backup](https://docs.photoprism.app/user-guide/backups/restore/#restoring-backups) documentation, which states that sidecar YAML files can be used to recover _some_ metadata. Are they read back then?

> some of the metadata and your albums can also be [recovered from YAML sidecar files](https://docs.photoprism.app/user-guide/backups/export/) even if you don't have a copy of the index database, unless you have [disabled this feature](https://docs.photoprism.app/getting-started/config-options/#feature-flags)

Which feature flag is permitting this?

**Reply by graciousgrey** [reply]
Metadata is read from YAML sidecar files if you index files for the first time. It won't be read if you index already indexed files again.

**Reply by tilusnet** [reply]
That's insightful @graciousgrey, thanks.

Does this also imply that if a file is deleted from PhotoPrism, importing it again will make read the sidecar files?

**Reply by lastzero** [reply]
Only if you remove the original, reindex with the cleanup option to permanently delete the index metadata, and then import the file again with the YAML as sidecar file (same folder and name, except for the extension). If you just want to add metadata, then exiftool compatible json sidecar files might be better suited!?

**Reply by tilusnet** [reply]
Oh it sounds like a heavy computational cost that way.

Okay I'll look into exiftool JSON files, thanks @lastzero. 
I assume I create one JSON file for each image file place it somewhere and PhotoPrism will merge the meta whenever it indexes / reindexes?

Is it this spec? https://exiftool.org/TagNames/JSON.html matched with https://www.photoprism.app/kb/metadata ?

**Comment by pbenaim** [comment]
So if I undertood that you said, I can delete the data base ( with:  **docker compose exec photoprism photoprism reset -y**  , with, before, **docker compose exec photoprism photoprism backup -a -i**  , just in case... )

And so after that, **docker compose exec photoprism photoprism index --cleanup** to remake index with yml (YAML) files ?

Say me if I'm wrong...

As a reminder, if it takes “a lot” of time it’s not a big deal. The important thing is to have the right information after modifying the YML files...

Thanks for your help 👍

**Comment by lastzero** [comment]
If you want to start with a clean database, just remove the database and recreate an empty one with the same name. Both the reset and the cleanup commands MAY remove the YAML sidecar files, so if you want to keep / use them, my recommendation is to be careful with that and carefully read the command descriptions e.g. using the --help flag.
